src/algorithms.jl

```diff
@@ -111,6 +111,15 @@ EK1(;
         initialization,
     )
 
+iskronecker(alg, f) = (
+    alg isa EK0
+    &&
+    !(alg.diffusionmodel isa DynamicMVDiffusion ||
+        alg.diffusionmodel isa FixedMVDiffusion)
+    && alg.prior isa IWP
+    && f.mass_matrix === I
```

Comment:
This could in principle be any `UniformScaling`, but make sure that the part that handles the mass matrix actually deals with this correctly.